models/tofu/bnb/tofu_bnb_events.sql

```diff
@@ -0,0 +1,135 @@
+           i as bundle_index
+    FROM (SELECT call_block_time,
+                 call_tx_hash,
+                 get_json_object(get_json_object(detail, '$.settlement'), '$.feeRate') / 1000000     as fee_rate,
+                 get_json_object(get_json_object(detail, '$.settlement'), '$.royaltyRate') / 1000000 as royalty_rate,
+                 get_json_object(get_json_object(detail, '$.settlement'), '$.feeAddress')            as fee_address,
+                 get_json_object(get_json_object(detail, '$.settlement'), '$.royaltyAddress')        as royalty_address,
+                 posexplode(from_json(get_json_object(detail, '$.bundle'), 'array<string>'))            as (i,t),
+                 json_array_length(get_json_object(detail, '$.bundle'))                              as bundle_size
+          FROM {{ source('tofu_nft_bnb', 'MarketNG_call_run') }}
+          WHERE call_success = true
+              {% if is_incremental() %}
+              and call_block_time >= date_trunc("day", now() - interval '1 week')
+              {% endif %}
+         ) as tmp
+),
+     tfe as (
+         select evt_tx_hash,
+                evt_block_time,
+                evt_block_number,
+                evt_index,
+                get_json_object(inventory, '$.seller')   as seller,
+                get_json_object(inventory, '$.buyer')    as buyer,
+                get_json_object(inventory, '$.kind')     as kind,
+                get_json_object(inventory, '$.price')    as price,
+                CASE WHEN get_json_object(inventory, '$.currency') = '0x0000000000000000000000000000000000000000'
+                  THEN '{{BNB_ERC20_ADDRESS}}'
+                  ELSE get_json_object(inventory, '$.currency')
+                END as currency,
+                (get_json_object(inventory, '$.currency') = '0x0000000000000000000000000000000000000000') as native_bnb,
+                contract_address
+         from {{ source('tofu_nft_bnb', 'MarketNG_evt_EvInventoryUpdate') }}
+         where get_json_object(inventory, '$.status') = '1'
+              {% if is_incremental() %}
+              and evt_block_time >= date_trunc("day", now() - interval '1 week')
+              {% endif %}
+     )
+SELECT 'bnb'                                 as blockchain
+     , 'tofu'                                as project
+     , 'v1'                                  as version
+     , date_trunc('day', tfe.evt_block_time) as block_date
+     , tfe.evt_block_time                    as block_time
+     , tfe.evt_block_number                  as block_number
+     , tff.token_id                          as token_id
+     , nft.standard                          as token_standard
+     , nft.name                              as collection
+     , case
+           when tff.bundle_size = 1 then 'Single Item Trade'
+           else 'Bundle Trade'
+    end                                      as trade_type
+     , tff.amount                            as number_of_items
```

Comment:
https://github.com/duneanalytics/spellbook/commit/b72d9e294dd7cdccc5330e1d417c319b10f8cb52#r94491148
I mentioned an issue about `nft.trades` on DuneSQL in another place.
Maybe we can patch a quickfix of this line, if this is the cause of the issue.
`, CAST(tff.amount AS DECIMAL(38,0))                 as number_of_items`

Reply:
Indeed - this needs to be explicitly casted to decimal, thank you @Hosuke!

Reply:
@Hosuke has kindly opened a PR to fix this <3 https://github.com/duneanalytics/spellbook/pull/2408